src/armory/Armory.js

```diff
@@ -157,39 +149,20 @@ class Armory extends React.Component<Props, State> {
 		setTankForCasus(newTank._id);
 	}
 
-	// Function that will save the selectedTank.
+	// Function that will save the selectedTank and update the user's inventory.
 	saveTank(): void {
-		const responsePromise: Promise<Response> = fetch('/api/tank/tankUpdate/' + this.state.selectedTank._id, {
-			method: 'PUT',
-			headers: {
-				'Access-Control-Allow-Origin': '*',
-				'Content-Type': 'application/json',
-				'Access-Control-Allow-Credentials': 'true',
-				'x-auth-token': getLoginToken(),
-			},
-			body: JSON.stringify({ 
-				tankName: this.state.selectedTank.tankName, 
-				userId: this.state.selectedTank.userId, 
-				components: this.state.selectedTank.parts.map(part => part.name),
-				isBot: false,
-			}),
+		updateTank(this.state.selectedTank, successful => {
+			if(successful) {
+				this.getUserInventory();
+			}
+			else {
+				toast.error('Could not save tank!');
```

Comment:
The error should probably be handled in the API call level

Reply:
Let me try if that's possible. I'm not sure if I can print toast messages from the API functions.

Reply:
Ok this works, gonna implement it.